disperser/batcher/batch_confirmer.go

```diff
@@ -0,0 +1,536 @@
+		}
+		if updateConfirmationInfoErr != nil {
+			b.logger.Error("error updating blob confirmed metadata", "err", updateConfirmationInfoErr)
+			blobsToRetry = append(blobsToRetry, batchData.blobs[blobIndex])
+		}
+	}
+
+	return blobsToRetry, nil
+}
+
+func (b *BatchConfirmer) ProcessConfirmedBatch(ctx context.Context, receiptOrErr *ReceiptOrErr) error {
+	if receiptOrErr.Metadata == nil {
+		return errors.New("failed to process confirmed batch: no metadata from transaction manager response")
+	}
+	confirmationMetadata := receiptOrErr.Metadata.(confirmationMetadata)
+	blobs := confirmationMetadata.blobs
+	if len(blobs) == 0 {
+		return errors.New("failed to process confirmed batch: no blobs from transaction manager metadata")
+	}
+	if receiptOrErr.Err != nil {
+		_ = b.handleFailure(ctx, blobs, FailConfirmBatch)
+		return fmt.Errorf("failed to confirm batch onchain: %w", receiptOrErr.Err)
+	}
+	if confirmationMetadata.aggSig == nil {
+		_ = b.handleFailure(ctx, blobs, FailNoAggregatedSignature)
+		return errors.New("failed to process confirmed batch: aggSig from transaction manager metadata is nil")
+	}
+	b.logger.Info("received ConfirmBatch transaction receipt", "blockNumber", receiptOrErr.Receipt.BlockNumber, "txnHash", receiptOrErr.Receipt.TxHash.Hex())
+
+	// Mark the blobs as complete
+	stageTimer := time.Now()
+	blobsToRetry, err := b.updateConfirmationInfo(ctx, confirmationMetadata, receiptOrErr.Receipt)
+	if err != nil {
+		_ = b.handleFailure(ctx, blobs, FailUpdateConfirmationInfo)
+		return fmt.Errorf("failed to update confirmation info: %w", err)
+	}
+	if len(blobsToRetry) > 0 {
+		b.logger.Error("failed to update confirmation info", "failed", len(blobsToRetry), "total", len(blobs))
+		_ = b.handleFailure(ctx, blobsToRetry, FailUpdateConfirmationInfo)
+	}
+	b.logger.Debug("Update confirmation info took", "duration", time.Since(stageTimer).String())
+	batchSize := int64(0)
+	for _, blobMeta := range blobs {
+		batchSize += int64(blobMeta.RequestMetadata.BlobSize)
+	}
+
+	return nil
+}
+
+func (b *BatchConfirmer) handleFailure(ctx context.Context, blobMetadatas []*disperser.BlobMetadata, reason FailReason) error {
+	var result *multierror.Error
+	numPermanentFailures := 0
+	for _, metadata := range blobMetadatas {
+		retry, err := b.BlobStore.HandleBlobFailure(ctx, metadata, b.MaxNumRetriesPerBlob)
+		if err != nil {
+			b.logger.Error("HandleSingleBatch: error handling blob failure", "err", err)
+			// Append the error
+			result = multierror.Append(result, err)
+		}
+
+		if retry {
+			continue
+		}
+
+		numPermanentFailures++
+	}
+
+	// Return the error(s)
+	return result.ErrorOrNil()
+}
+
+func (b *BatchConfirmer) HandleSingleBatch(ctx context.Context) error {
+	currentBlock, err := b.ethClient.BlockNumber(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting current block number: %w", err)
+	}
+
+	err = b.EncodingStreamer.UpdateReferenecBlock(uint(currentBlock))
```

Comment:
nit: there seems to be a typo in the name of this function

Reply:
nice catch